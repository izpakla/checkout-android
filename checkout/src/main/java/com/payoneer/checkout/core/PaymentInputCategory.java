/*
 * Copyright (c) 2020 Payoneer Germany GmbH
 * https://www.payoneer.com
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package com.payoneer.checkout.core;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.StringDef;

/**
 * Class containing the different payment input categories i.e. inputelement, extraelement or registration values.
 */
public class PaymentInputCategory {
    public final static String INPUTELEMENT = "inputelement";
    public final static String EXTRAELEMENT = "extraelement";
    public final static String REGISTRATION = "registration";

    /**
     * Check if the given category is a valid payment input value category
     *
     * @param category the payment input category to validate
     * @return true when valid, false otherwise
     */
    public static boolean isValid(final String category) {

        if (category != null) {
            switch (category) {
                case INPUTELEMENT:
                case EXTRAELEMENT:
                case REGISTRATION:
                    return true;
            }
        }
        return false;
    }

    @Retention(RetentionPolicy.SOURCE)
    @StringDef({
        INPUTELEMENT,
        EXTRAELEMENT,
        REGISTRATION
    })
    public @interface Definition { }
}
