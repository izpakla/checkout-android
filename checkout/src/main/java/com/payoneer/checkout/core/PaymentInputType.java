/*
 * Copyright (c) 2020 Payoneer Germany GmbH
 * https://www.payoneer.com
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package com.payoneer.checkout.core;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.StringDef;

/**
 * Class containing the different payment input types i.e. number, iban, bic, allowRecurrence etc.
 */
public class PaymentInputType {
    public final static String HOLDER_NAME = "holderName";
    public final static String ACCOUNT_NUMBER = "number";
    public final static String BANK_CODE = "bankCode";
    public final static String BANK_NAME = "bankName";
    public final static String BIC = "bic";
    public final static String BRANCH = "branch";
    public final static String CITY = "city";
    public final static String EXPIRY_MONTH = "expiryMonth";
    public final static String EXPIRY_YEAR = "expiryYear";
    public final static String EXPIRY_DATE = "expiryDate";
    public final static String IBAN = "iban";
    public final static String LOGIN = "login";
    public final static String OPTIN = "optIn";
    public final static String PASSWORD = "password";
    public final static String VERIFICATION_CODE = "verificationCode";
    public final static String CUSTOMER_BIRTHDAY = "customerBirthDay";
    public final static String CUSTOMER_BIRTHMONTH = "customerBirthMonth";
    public final static String CUSTOMER_BIRTHYEAR = "customerBirthYear";
    public final static String INSTALLMENT_PLANID = "installmentPlanId";
    public final static String ALLOW_RECURRENCE = "allowRecurrence";
    public final static String AUTO_REGISTRATION = "autoRegistration";
    public final static String REGISTRATION_OPTIONS = "registrationOptions";

    /**
     * Check if the given type is a valid payment input type
     *
     * @param type the payment input type to validate
     * @return true when valid, false otherwise
     */
    public static boolean isValid(final String type) {

        if (type != null) {
            switch (type) {
                case HOLDER_NAME:
                case ACCOUNT_NUMBER:
                case BANK_CODE:
                case BANK_NAME:
                case BIC:
                case BRANCH:
                case CITY:
                case EXPIRY_MONTH:
                case EXPIRY_YEAR:
                case EXPIRY_DATE:
                case IBAN:
                case LOGIN:
                case OPTIN:
                case PASSWORD:
                case VERIFICATION_CODE:
                case CUSTOMER_BIRTHDAY:
                case CUSTOMER_BIRTHMONTH:
                case CUSTOMER_BIRTHYEAR:
                case INSTALLMENT_PLANID:
                case ALLOW_RECURRENCE:
                case AUTO_REGISTRATION:
                    return true;
            }
        }
        return false;
    }

    @Retention(RetentionPolicy.SOURCE)
    @StringDef({
        HOLDER_NAME,
        ACCOUNT_NUMBER,
        BANK_CODE,
        BANK_NAME,
        BIC,
        BRANCH,
        CITY,
        EXPIRY_MONTH,
        EXPIRY_YEAR,
        EXPIRY_DATE,
        IBAN,
        LOGIN,
        OPTIN,
        PASSWORD,
        VERIFICATION_CODE,
        CUSTOMER_BIRTHDAY,
        CUSTOMER_BIRTHMONTH,
        CUSTOMER_BIRTHYEAR,
        INSTALLMENT_PLANID,
        ALLOW_RECURRENCE,
        AUTO_REGISTRATION
    })
    public @interface Definition { }
}
