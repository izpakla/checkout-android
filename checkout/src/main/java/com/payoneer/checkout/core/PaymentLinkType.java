/*
 * Copyright (c) 2020 Payoneer Germany GmbH
 * https://www.payoneer.com
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package com.payoneer.checkout.core;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.StringDef;

/**
 * Class containing the different types of links
 */
public class PaymentLinkType {

    public final static String SELF = "self";
    public final static String ONSELECT = "onSelect";
    public final static String LOGO = "logo";
    public final static String OPERATION = "operation";
    public final static String LANGUAGE = "lang";

    /**
     * Check if the given type is a valid payment link
     *
     * @param type the payment input type to validate
     * @return true when valid, false otherwise
     */
    public static boolean isValid(final String type) {

        if (type != null) {
            switch (type) {
                case SELF:
                case ONSELECT:
                case LOGO:
                case OPERATION:
                case LANGUAGE:
                    return true;
            }
        }
        return false;
    }

    @Retention(RetentionPolicy.SOURCE)
    @StringDef({
        SELF,
        ONSELECT,
        LOGO,
        OPERATION,
        LANGUAGE
    })
    public @interface Definition { }
}
