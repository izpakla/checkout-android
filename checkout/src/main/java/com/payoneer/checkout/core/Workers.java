/*
 * Copyright (c) 2020 Payoneer Germany GmbH
 * https://www.payoneer.com
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package com.payoneer.checkout.core;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

/**
 * The Singleton class maintaining a list of Workers processing background tasks
 */
public final class Workers {

    private final ExecutorService networkService;

    private Workers() {
        this.networkService = Executors.newCachedThreadPool();
    }

    /**
     * Get the instance of this Workers
     *
     * @return the instance of this Workers
     */
    public static Workers getInstance() {
        return InstanceHolder.INSTANCE;
    }

    /**
     * Get the WorkerExecutor for network tasks
     *
     * @return the network executor for network tasks
     */
    public ExecutorService forNetworkTasks() {
        return networkService;
    }

    private static class InstanceHolder {
        static final Workers INSTANCE = new Workers();
    }
}
