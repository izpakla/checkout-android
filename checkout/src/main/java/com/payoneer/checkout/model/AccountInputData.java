/*
 * Copyright (c) 2020 Payoneer Germany GmbH
 * https://www.payoneer.com
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package com.payoneer.checkout.model;

/**
 * Input data what could been submitted by payment-page form.
 */
public class AccountInputData {
    /** Simple API, optional */
    private String holderName;
    /** Simple API, optional */
    private String number;
    /** Simple API, optional */
    private String bankCode;
    /** Simple API, optional */
    private String bankName;
    /** Simple API, optional */
    private String bic;
    /** Simple API, optional */
    private String branch;
    /** Simple API, optional */
    private String city;
    /** Simple API, optional */
    private String expiryMonth;
    /** Simple API, optional */
    private String expiryYear;
    /** Simple API, optional */
    private String iban;
    /** Simple API, optional */
    private String login;
    /** Simple API, optional */
    private Boolean optIn;
    /** Simple API, optional */
    private String password;
    /** Simple API, optional */
    private String verificationCode;

    /** day of customer's birthday */
    private String customerBirthDay;
    /** month of customer's birthday */
    private String customerBirthMonth;
    /** year of customer's birthday */
    private String customerBirthYear;

    /** id of installment plan */
    private String installmentPlanId;

    public String getHolderName() {
        return holderName;
    }

    public void setHolderName(final String holderName) {
        this.holderName = holderName;
    }

    public String getNumber() {
        return number;
    }

    public void setNumber(final String number) {
        this.number = number;
    }

    public String getBankCode() {
        return bankCode;
    }

    public void setBankCode(final String bankCode) {
        this.bankCode = bankCode;
    }

    public String getBankName() {
        return bankName;
    }

    public void setBankName(final String bankName) {
        this.bankName = bankName;
    }

    public String getBic() {
        return bic;
    }

    public void setBic(final String bic) {
        this.bic = bic;
    }

    public String getBranch() {
        return branch;
    }

    public void setBranch(final String branch) {
        this.branch = branch;
    }

    public String getCity() {
        return city;
    }

    public void setCity(final String city) {
        this.city = city;
    }

    public String getExpiryMonth() {
        return expiryMonth;
    }

    public void setExpiryMonth(final String expiryMonth) {
        this.expiryMonth = expiryMonth;
    }

    public String getExpiryYear() {
        return expiryYear;
    }

    public void setExpiryYear(final String expiryYear) {
        this.expiryYear = expiryYear;
    }

    public String getIban() {
        return iban;
    }

    public void setIban(final String iban) {
        this.iban = iban;
    }

    public String getLogin() {
        return login;
    }

    public void setLogin(final String login) {
        this.login = login;
    }

    public Boolean getOptIn() {
        return optIn;
    }

    public void setOptIn(final Boolean optIn) {
        this.optIn = optIn;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(final String password) {
        this.password = password;
    }

    public String getVerificationCode() {
        return verificationCode;
    }

    public void setVerificationCode(final String verificationCode) {
        this.verificationCode = verificationCode;
    }

    public String getCustomerBirthDay() {
        return customerBirthDay;
    }

    public void setCustomerBirthDay(final String customerBirthDay) {
        this.customerBirthDay = customerBirthDay;
    }

    public String getCustomerBirthMonth() {
        return customerBirthMonth;
    }

    public void setCustomerBirthMonth(final String customerBirthMonth) {
        this.customerBirthMonth = customerBirthMonth;
    }

    public String getCustomerBirthYear() {
        return customerBirthYear;
    }

    public void setCustomerBirthYear(final String customerBirthYear) {
        this.customerBirthYear = customerBirthYear;
    }

    public String getInstallmentPlanId() {
        return installmentPlanId;
    }

    public void setInstallmentPlanId(final String installmentPlanId) {
        this.installmentPlanId = installmentPlanId;
    }

    @Override
    public String toString() {
        final StringBuilder builder = new StringBuilder();
        builder.append("AccountInputData [");
        if (holderName != null) {
            builder.append("holderName=").append(holderName).append(", ");
        }
        if (number != null) {
            builder.append("number=").append(number).append(", ");
        }
        if (bankCode != null) {
            builder.append("bankCode=").append(bankCode).append(", ");
        }
        if (bankName != null) {
            builder.append("bankName=").append(bankName).append(", ");
        }
        if (bic != null) {
            builder.append("bic=").append(bic).append(", ");
        }
        if (branch != null) {
            builder.append("branch=").append(branch).append(", ");
        }
        if (city != null) {
            builder.append("city=").append(city).append(", ");
        }
        if (expiryMonth != null) {
            builder.append("expiryMonth=").append(expiryMonth).append(", ");
        }
        if (expiryYear != null) {
            builder.append("expiryYear=").append(expiryYear).append(", ");
        }
        if (iban != null) {
            builder.append("iban=").append(iban).append(", ");
        }
        if (login != null) {
            builder.append("login=").append(login).append(", ");
        }
        if (optIn != null) {
            builder.append("optIn=").append(optIn).append(", ");
        }
        if (password != null) {
            builder.append("password=").append(password).append(", ");
        }
        if (verificationCode != null) {
            builder.append("verificationCode=").append(verificationCode).append(", ");
        }
        if (customerBirthDay != null) {
            builder.append("customerBirthDay=").append(customerBirthDay).append(", ");
        }
        if (customerBirthMonth != null) {
            builder.append("customerBirthMonth=").append(customerBirthMonth).append(", ");
        }
        if (customerBirthYear != null) {
            builder.append("customerBirthYear=").append(customerBirthYear).append(", ");
        }
        if (installmentPlanId != null) {
            builder.append("installmentPlanId=").append(installmentPlanId);
        }
        builder.append("]");
        return builder.toString();
    }
}
