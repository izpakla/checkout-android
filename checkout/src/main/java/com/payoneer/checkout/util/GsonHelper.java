/*
 * Copyright (c) 2020 Payoneer Germany GmbH
 * https://www.payoneer.com
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package com.payoneer.checkout.util;

import java.lang.reflect.Type;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonSyntaxException;

import android.text.TextUtils;

/**
 * Class with helper methods for Gson
 */
public final class GsonHelper {

    private final Gson gson;

    private GsonHelper() {
        this.gson = new GsonBuilder().create();
    }

    /**
     * Get the instance of this GsonHelper
     *
     * @return the instance of this GsonHelper
     */
    public static GsonHelper getInstance() {
        return InstanceHolder.INSTANCE;
    }

    /**
     * This method serializes the specified object into its equivalent Json representation.
     *
     * @return Json representation of src
     */
    public String toJson(Object src) {
        return src == null ? null : gson.toJson(src);
    }

    public <T> T fromJson(String json, Class<T> classOfT) throws JsonSyntaxException {
        return TextUtils.isEmpty(json) ? null : gson.fromJson(json, classOfT);
    }

    public <T> T fromJson(String json, Type type) throws JsonSyntaxException {
        return TextUtils.isEmpty(json) ? null : gson.fromJson(json, type);
    }

    private static class InstanceHolder {
        static final GsonHelper INSTANCE = new GsonHelper();

    }
}
