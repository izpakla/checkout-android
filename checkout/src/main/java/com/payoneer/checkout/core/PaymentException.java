/*
 * Copyright (c) 2020 Payoneer Germany GmbH
 * https://www.payoneer.com
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package com.payoneer.checkout.core;

import com.payoneer.checkout.model.ErrorInfo;

/**
 * The PaymentException containing the details of the payment error
 */
public class PaymentException extends Exception {

    /** The error info obtained from the payment API */
    private ErrorInfo errorInfo;

    /** Indicates that the exception is caused by a network failure, e.g. poor wifi connection */
    private boolean networkFailure;

    /**
     * {@inheritDoc}
     *
     * @param message the extra error info
     */
    public PaymentException(final String message) {
        super(message);
    }

    /**
     * {@inheritDoc}
     *
     * @param cause the cause of this exception
     */
    public PaymentException(final Throwable cause) {
        super(cause.getMessage(), cause);
    }

    /**
     * {@inheritDoc}
     *
     * @param message containing a description of the error
     * @param cause of the error
     */
    public PaymentException(final String message, final Throwable cause) {
        super(message, cause);
    }

    /**
     * Constructs a new PaymentException with the cause and networkFailure flag
     *
     * @param cause of the error
     * @param networkFailure indicates that the exception was caused by a network failure
     */
    public PaymentException(final Throwable cause, boolean networkFailure) {
        super(cause);
        this.networkFailure = networkFailure;
    }

    /**
     * Constructs a new PaymentException
     *
     * @param errorInfo information about the error
     */
    public PaymentException(final ErrorInfo errorInfo) {
        super(errorInfo.getResultInfo());
        this.errorInfo = errorInfo;
    }

    public ErrorInfo getErrorInfo() {
        return errorInfo;
    }

    public boolean getNetworkFailure() {
        return networkFailure;
    }
}
