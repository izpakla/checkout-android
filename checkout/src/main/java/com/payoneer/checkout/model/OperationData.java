/*
 * Copyright (c) 2020 Payoneer Germany GmbH
 * https://www.payoneer.com
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package com.payoneer.checkout.model;

import java.util.List;
import java.util.Map;

/**
 * This class is designed to hold information for operation (CHARGE, PAYOUT, UPDATE) with selected payment network.
 */
public class OperationData {
    /** Simple API, optional */
    private AccountInputData account;
    /** Simple API, optional */
    private Boolean autoRegistration;
    /** Simple API, optional */
    private Boolean allowRecurrence;
    /** Advanced API, optional */
    private Map<String, Boolean> checkboxes;
    /** Provider request parameters. */
    private ProviderParameters providerRequest;
    /** List of Provider request parameters. */
    private List<ProviderParameters> providerRequests;
    /** Customer web browser data */
    private BrowserData browserData;

    public AccountInputData getAccount() {
        return account;
    }

    public void setAccount(final AccountInputData account) {
        this.account = account;
    }

    public Boolean getAutoRegistration() {
        return autoRegistration;
    }

    public void setAutoRegistration(final Boolean autoRegistration) {
        this.autoRegistration = autoRegistration;
    }

    public Boolean getAllowRecurrence() {
        return allowRecurrence;
    }

    public void setAllowRecurrence(final Boolean allowRecurrence) {
        this.allowRecurrence = allowRecurrence;
    }

    public Map<String, Boolean> getCheckboxes() {
        return checkboxes;
    }

    public void setCheckboxes(final Map<String, Boolean> checkboxes) {
        this.checkboxes = checkboxes;
    }

    public ProviderParameters getProviderRequest() {
        return providerRequest;
    }

    public void setProviderRequest(final ProviderParameters providerRequest) {
        this.providerRequest = providerRequest;
    }

    public List<ProviderParameters> getProviderRequests() {
        return providerRequests;
    }

    public void setProviderRequests(final List<ProviderParameters> providerRequests) {
        this.providerRequests = providerRequests;
    }

    public BrowserData getBrowserData() {
        return browserData;
    }

    public void setBrowserData(final BrowserData browserData) {
        this.browserData = browserData;
    }


    @Override
    public String toString() {
        final StringBuilder builder = new StringBuilder();
        builder.append("OperationData [");
        if (account != null) {
            builder.append("account=").append(account).append(", ");
        }
        if (autoRegistration != null) {
            builder.append("autoRegistration=").append(autoRegistration).append(", ");
        }
        if (allowRecurrence != null) {
            builder.append("allowRecurrence=").append(allowRecurrence).append(", ");
        }
        if (checkboxes != null) {
            builder.append("checkboxes=").append(checkboxes).append(", ");
        }
        if (providerRequest != null) {
            builder.append("providerRequest=").append(providerRequest).append(", ");
        }
        if (providerRequests != null) {
            builder.append("providerRequests=").append(providerRequests).append(", ");
        }
        if (browserData != null) {
            builder.append("browserData=").append(browserData);
        }
        builder.append("]");
        return builder.toString();
    }
}
