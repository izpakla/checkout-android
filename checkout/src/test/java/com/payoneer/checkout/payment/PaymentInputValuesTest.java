/*
 * Copyright (c) 2022 Payoneer Germany GmbH
 * https://www.payoneer.com
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package com.payoneer.checkout.payment;

import static io.github.jsonSnapshot.SnapshotMatcher.expect;
import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static io.github.jsonSnapshot.SnapshotMatcher.validateSnapshots;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import com.payoneer.checkout.core.PaymentInputCategory;
import com.payoneer.checkout.core.PaymentInputType;
import com.payoneer.checkout.model.AccountInputData;
import com.payoneer.checkout.model.OperationData;
import com.payoneer.checkout.util.GsonHelper;

@RunWith(RobolectricTestRunner.class)
public class PaymentInputValuesTest {

    @BeforeClass
    public static void beforeClass() {
        start();
    }

    @AfterClass
    public static void afterAll() {
        validateSnapshots();
    }

    @Test(expected = IllegalArgumentException.class)
    public void putValue_invalidCategory_exception() {
        PaymentInputValues values = new PaymentInputValues();
        values.putStringValue(null, PaymentInputType.HOLDER_NAME, "Foo");
    }

    @Test(expected = IllegalArgumentException.class)
    public void putValue_invalidName_exception() {
        PaymentInputValues values = new PaymentInputValues();
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, null, "Foo");
    }

    @Test
    public void copyInto() {
        PaymentInputValues values = new PaymentInputValues();
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.HOLDER_NAME, "John Doe");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.ACCOUNT_NUMBER, "accountnumber123");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.BANK_CODE, "bankcode123");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.BANK_NAME, "bankname123");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.BIC, "bic123");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.BRANCH, "branch123");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.CITY, "city123");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.EXPIRY_MONTH, "12");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.EXPIRY_YEAR, "2019");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.IBAN, "iban123");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.LOGIN, "login123");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.PASSWORD, "password123");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.VERIFICATION_CODE, "123");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.CUSTOMER_BIRTHDAY, "3");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.CUSTOMER_BIRTHMONTH, "12");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.CUSTOMER_BIRTHYEAR, "72");
        values.putStringValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.INSTALLMENT_PLANID, "72");
        values.putBooleanValue(PaymentInputCategory.INPUTELEMENT, PaymentInputType.OPTIN, true);
        values.putBooleanValue(PaymentInputCategory.REGISTRATION, PaymentInputType.ALLOW_RECURRENCE, true);
        values.putBooleanValue(PaymentInputCategory.REGISTRATION, PaymentInputType.AUTO_REGISTRATION, true);

        OperationData operationData = new OperationData();
        operationData.setAccount(new AccountInputData());

        values.copyInto(operationData);
        String json = GsonHelper.getInstance().toJson(operationData);
        expect(json).toMatchSnapshot();
    }
}

