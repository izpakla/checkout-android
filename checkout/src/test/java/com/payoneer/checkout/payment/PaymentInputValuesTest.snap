com.payoneer.checkout.payment.PaymentInputValuesTest.copyInto=[
  "{\"account\":{\"holderName\":\"John Doe\",\"number\":\"accountnumber123\",\"bankCode\":\"bankcode123\",\"bankName\":\"bankname123\",\"bic\":\"bic123\",\"branch\":\"branch123\",\"city\":\"city123\",\"expiryMonth\":\"12\",\"expiryYear\":\"2019\",\"iban\":\"iban123\",\"login\":\"login123\",\"optIn\":true,\"password\":\"password123\",\"verificationCode\":\"123\",\"customerBirthDay\":\"3\",\"customerBirthMonth\":\"12\",\"customerBirthYear\":\"72\",\"installmentPlanId\":\"72\"},\"autoRegistration\":true,\"allowRecurrence\":true}"
]